modules/account-lib/test/unit/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -81,6 +83,32 @@ describe('TSS ECDSA TESTS', function () {
     pallierMock.restore();
   });
 
+  describe('rangeproof tests', function () {
+    const NODE_MAJOR_VERSION = parseInt(process.versions.node.split('.')[0], 10);
+    it('should generate safe primes of a specified bitlength', async function () {
+      const bitlength = 128; // this should be small enough to be reasonably fast
+      if (NODE_MAJOR_VERSION === NaN || NODE_MAJOR_VERSION <= 14) {
+        await assert.rejects(rangeProof.generateSafePrime(bitlength));
+      } else {
+        const p = await rangeProof.generateSafePrime(bitlength);
+        (await isProbablyPrime(p)).should.be.true();
+        const q = (p - BigInt(1)) / BigInt(2);
+        (await isProbablyPrime(q)).should.be.true();
+        bitLength(p).should.equal(bitlength);
```

Comment:
I may be missing something here, But does q always resolves to a prime number given p is prime? (eg: when p =17)